tests/auxiliary/user-account/test_contract.py

```diff
@@ -0,0 +1,372 @@
+    assert get_nonce(vm) == 2
+
+
+def test_call_checks_nonce(vm):
+    transaction = ForwardingTransaction(**{**DEFAULT_TX_PARAMS, **{
+        "nonce": 0,
+    }}).sign(PRIVATE_KEY)
+    computation = vm.apply_transaction(transaction)
+    assert computation.is_success
+
+    computation = vm.apply_transaction(transaction)
+    assert computation.is_error
+
+    transaction = ForwardingTransaction(**{**DEFAULT_TX_PARAMS, **{
+        "nonce": 2,
+    }}).sign(PRIVATE_KEY)
+    computation = vm.apply_transaction(transaction)
+    assert computation.is_error
+
+
+def test_call_checks_block_range(vm):
+    assert vm.block.number == 10
+
+    invalid_block_ranges = [
+        (0, 9),
+        (5, 9),
+        (11, 20),
+        (11, UINT_256_MAX),
+        (11, 9),
+        (UINT_256_MAX, 0),
+    ]
+    for min_block, max_block in invalid_block_ranges:
+        transaction = ForwardingTransaction(**{**DEFAULT_TX_PARAMS, **{
+            "nonce": get_nonce(vm),
+            "min_block": min_block,
+            "max_block": max_block,
+        }}).sign(PRIVATE_KEY)
+        computation = vm.apply_transaction(transaction)
+        assert computation.is_error
+
+    valid_block_ranges = [
+        (0, UINT_256_MAX),
+        (0, 10),
+        (10, 10),
+        (10, UINT_256_MAX)
+    ]
+    for min_block, max_block in valid_block_ranges:
+        transaction = ForwardingTransaction(**{**DEFAULT_TX_PARAMS, **{
+            "nonce": get_nonce(vm),
+            "min_block": min_block,
+            "max_block": max_block,
+        }}).sign(PRIVATE_KEY)
+        computation = vm.apply_transaction(transaction)
+        assert computation.is_success
+
+
+def test_call_transfers_value(vm):
+    with vm.state.state_db() as state_db:
+        balance_sender_before = state_db.get_balance(ACCOUNT_ADDRESS)
+        balance_destination_before = state_db.get_balance(DESTINATION_ADDRESS)
+
+    transaction = ForwardingTransaction(**{**DEFAULT_TX_PARAMS, **{
+        "nonce": get_nonce(vm),
+        "value": 10
+    }}).sign(PRIVATE_KEY)
+    computation = vm.apply_transaction(transaction)
+    assert computation.is_success
+
+    with vm.state.state_db() as state_db:
+        balance_sender_after = state_db.get_balance(ACCOUNT_ADDRESS)
+        balance_destination_after = state_db.get_balance(DESTINATION_ADDRESS)
+
+    assert balance_sender_after == balance_sender_before - 10
+    assert balance_destination_after == balance_destination_before + 10
+
+
+def test_call_checks_signature(vm):
+    transaction = ForwardingTransaction(**{**DEFAULT_TX_PARAMS, **{
+        "nonce": get_nonce(vm),
+    }}).sign(PRIVATE_KEY)
+
+    v, r, s = transaction.vrs
+    invalid_vrs_triples = [
+        (v + 1, r, s),
+        (v - 1, r, s),
+        (0, r, s),
+        (1, r, s),
+
+        (v, r + 1, s),
+        (v, r - 1, s),
+        (v, 0, s),
+
+        (v, r, s + 1),
+        (v, r, s - 1),
+        (v, r, 0),
+
+        (27 if v == 28 else 28, r, SECPK1_N - s)
```

Comment:
can we parametrize these using `pytest.mark.parametrize` so that they each run in isolation?